docs/ImageMirrorSetDesign.md

```diff
@@ -0,0 +1,167 @@
+# Summary
+
+Users need a way to set up mirror configuration. Users can set mirror configurations using ImageContentSourcePolicy currently via MCO currently, but if they want to use mirror when pulling image with tag instead of digest in the pull spec, they need to follow this [solution](https://access.redhat.com/solutions/4817401) to update the registries configuration via MCO. We can hopefully help customers by documenting CRDs with the `ImageDigestMirrorSet` and `ImageTagMirrorSet` tuneables and have the MCO use these tuneables when rendering the registries.conf files to Ignition/disk.
+
+# Motivation
+
+The MCO contains the logic to write the registries.conf configuration files to Ignition. When Ignition starts on a machine it writes this file to configure the mirrors.
+
+## Goals
+
+1. Setting cluster wide mirror registries via `ImageDigestMirrorSet`. The mirrors will be used only when the pull spec is specified with a digest.
+
+2. Setting cluster wide mirror registries via `ImageTagMirrorSet`. The mirrors will be used only when the pull spec is specified with a tag.
+
+3. Setting cluster wide mirror registries, but blocking the image pull from the registry that specified as source in the pull spec by setting the enum field `mirrorSourcePolicy`. If all the mirror registries are failed, the image pull will be redirected to the registry in the pull spec by default.
```

Comment:
Is mirrorSourcePolicy available for both ITMS and IDMS? At a glance in the MCO codebase only IDMS seems to have this field

Reply:
update that mirrorSourcePolicy is available for both